### ISSUE TITLE
phemex Spot positions / get average entry price ?

### ISSUE THREAD

**Issue by adamrg73** [ccxt/ccxt#12620]
exchange.fetch_positions()
doesnt work for Spot symbols.

How do we get the average entry price for spot positions without writing a custom function to check all orders, and amount ordered and amount currently in wallet etc.
I have an idea of how to write the custom function, but concerned I may get it wrong considering what I just mentioned above, for example, if you move BTC to the contract account in phemex, it won't show it was sold, so then if you check all orders to check average entry price, it would be wrong. Know what I mean?

**Comment by adamrg73** [comment]
<img width="486" alt="Screen Shot 2022-04-04 at 5 27 36 PM" src="https://user-images.githubusercontent.com/70821026/161635162-31324970-4617-43ba-8bc6-76ee0e7014a7.png">


this is what phemex API is saying regarding this issue.

**Comment by adamrg73** [comment]
looks like this is a new endpoint, the other ones deprecated.
https://github.com/phemex/phemex-api-docs/blob/master/Public-Spot-API-en.md#spotQueryClosedOrder
says deprecated

**Comment by adamrg73** [comment]
i tried changing all these endpoints, and fetch_closed_orders still using Response: GET https://api.phemex.com/exchange/spot/order?symbol=sDAIUSDT 200


print('exhange.api:')
        print(exchange.api)
        print(exchange.api['private']['post'][0])
        exchange.api['private']['post'][0]='api-data/spots/orders'
        print(exchange.api['private']['post'][0])
        print(exchange.api['private']['post'][1])
        exchange.api['private']['post'][1]='api-data/spots/orders'
        print(exchange.api['private']['post'][1])
        print(exchange.api['private']['put'][0])
        exchange.api['private']['put'][0]='api-data/spots/orders'
        print(exchange.api['private']['put'][0])
        print(exchange.api['private']['get'][1])
        exchange.api['private']['get'][1]='api-data/spots/orders'
        print(exchange.api['private']['get'][1])
        print(exchange.api['private']['get'][3])
        exchange.api['private']['get'][3]='api-data/spots/orders'
        print(exchange.api['private']['get'][3])
        print(exchange.api['private']['get'][3])
        exchange.api['private']['get'][3]='api-data/spots/orders'
        print(exchange.api['private']['get'][3])
        print(exchange.api['private']['get'][4])
        exchange.api['private']['get'][4]='api-data/spots/orders'
        print(exchange.api['private']['get'][4])
        print(exchange.api['private']['get'][0])
        exchange.api['private']['get'][0]='api-data/spots/orders'
        print(exchange.api['private']['get'][0])
        print(exchange.api['private']['get'][8])
        exchange.api['private']['get'][8]='api-data/spots/orders'
        print(exchange.api['private']['get'][8])
        print(exchange.api['private']['get'][9])
        exchange.api['private']['get'][9]='api-data/spots/orders'
        print(exchange.api['private']['get'][9])
        print(exchange.api['private']['get'][10])
        exchange.api['private']['get'][10]='api-data/spots/orders'
        print(exchange.api['private']['get'][10])
        print(exchange.api['private']['get'][11])
        exchange.api['private']['get'][11]='api-data/spots/orders'
        print(exchange.api['private']['get'][11])
        print(exchange.api['private']['delete'][0])
        exchange.api['private']['delete'][0]='api-data/spots/orders'
        print(exchange.api['private']['delete'][0])
        print(exchange.api['private']['delete'][1])
        exchange.api['private']['delete'][1]='api-data/spots/orders'
        print(exchange.api['private']['delete'][1])
        print(exchange.api)

**Comment by Dan-krm** [comment]
Hi, fetchPositions isn't meant for spot orders. The method fetchClosedOrders uses the endpoint that Phemex recommends. CCXT currently doesn't have a standalone method that determines your average entry price for spot positions.

My recommendation would be to use fetchClosedOrders() and create a custom function that calculates the overall weighting for each order and using the average value, find the average entry price.
```
phemex.fetchClosedOrders (BTC/USDT)
2022-05-17T23:12:02.755Z iteration 0 passed in 544 ms

                                  id |                        clientOrderId |     timestamp |                 datetime | lastTradeTimestamp |   symbol |   type | timeInForce | postOnly | side |    price | stopPrice |  amount |       cost |  average |   filled | remaining | status |             fee | trades |              fees
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
8dcb3a4e-00cf-4ddd-ac8b-3eb26b4f2423 | f5833d61-651b-5081-c497-9277fdae7052 | 1652828805061 | 2022-05-17T23:06:45.061Z |                    | BTC/USDT | market |         IOC |    false |  buy | 33548.39 |           | 0.00034 | 10.3695036 | 30498.54 |  0.00034 |         0 | closed | {"cost":3.4e-7} |     [] | [{"cost":3.4e-7}]
faa03459-7c54-4e68-8d3e-7e7fdfe70f86 | d4305fe2-9ccd-5461-ff43-b0e6ff11ef71 | 1652829107331 | 2022-05-17T23:11:47.331Z |                    | BTC/USDT | market |         IOC |    false |  buy | 33525.94 |           | 0.00034 | 10.3641486 | 30482.79 |  0.00034 |         0 | closed | {"cost":3.4e-7} |     [] | [{"cost":3.4e-7}]
2 objects
```

To handle your last comment about sending your amount to contract and not showing it as sold, I've opened a PR for a fetchTransfers method that can be used to track this in your custom function if you'd like. You could use the timestamp from fetchTransfers with fetchOHLCV to find the value of the transfer when it happened.
```
phemex.fetchTransfers (BTC)
2022-05-18T00:11:29.908Z iteration 0 passed in 558 ms

                                  id |     timestamp |                 datetime | currency |     amount | fromAccount | toAccount | status
------------------------------------------------------------------------------------------------------------------------------------------
87c071a3-8628-4ac2-aca1-6ce0d1fad66c | 1652832467000 | 2022-05-18T00:07:47.000Z |      BTC | 0.00067932 |        spot |    future |     ok
1 objects
```

Phemex also has this conversions endpoint if you've done any conversions through the Phemex interface that you would like to track: https://github.com/phemex/phemex-api-docs/blob/master/Public-Transfer-API-en.md#query-convert-history There currently isn't a unified method on CCXT that supports this type of endpoint because most exchanges don't have this specific type of functionality.
